### DISCUSSION TITLE
Authentication on Nextjs middleware

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#518] [Help]
I am using web3auth + auth0 for user authentication
Now, if a user is not authenticated or if the user tries to navigate to a different url which the application does not support, the the user should be redirected to the signin page
For that, I want to use the middleware in nextjs application. 
Does web3Auth provide any middleware to achieve this functionality?

**Comment by shahbaz17** [comment]
Hey @KetulCodiste 

Web3Auth doesn't provide such middleware, but you can create your own middleware and use the web3auth state to check the log-in states and redirect users accordingly.

**Comment by yashovardhan** [comment]
You can subscribe to [Web3Auth Lifecycle Events](https://web3auth.io/docs/sdk/web/web3auth/initialize#subscribing-the-lifecycle-events) to verify if the user is logged in or not. You can configure your redirection URLs accordingly.
